File: 프로그래머스/python/Level 2/모음 사전.py
def solution(word):
    word_dict = []

    def make_word(curWord):
        if len(curWord) != 0: 
            word_dict.append(curWord)
        if len(curWord) == 5:
            return
        
        vowel = ['A', 'E', 'I', 'O', 'U']
        for v in vowel:
            make_word(curWord+v)
            
    make_word("")
    return word_dict.index(word) + 1

File: 프로그래머스/js/Level 2/모음사전.js
function solution(word) {
  word_dict = [];
  function make_word(curWord) {
    if (curWord.length != 0) word_dict.push(curWord);
    if (curWord.length == 5) return;

    const vowel = ["A", "E", "I", "O", "U"];
    for (let v of vowel) {
      make_word(curWord + v);
    }
  }
  make_word("");
  return word_dict.indexOf(word) + 1;
}
